docs/npm.md

```diff
@@ -0,0 +1,395 @@
+  * See [package.json](https://docs.npmjs.com/cli/v9/configuring-npm/package-json)
+  * See [package-lock.json](https://docs.npmjs.com/cli/v9/configuring-npm/package-lock-json)
+
+The package-lock.json version must be **higher than v1**! package-lock.json v1 is not supported in Cachi2.
+The last npm version associated with package-lock.json v1 was in nodejs v14 (which was shipped with npm version 6), which is EOL since 30 Apr 2023.
+See [nodejs endoflife.date](https://endoflife.date/nodejs).
+
+Cachi2 fetch-deps shell command:
+
+```shell
+cachi2 fetch-deps \
+  --source ./my-repo \
+  --output ./cachi2-output \
+  '<JSON input>'
+```
+
+JSON input:
+```jsonc
+{
+  "type": "npm",
+  // path to the package (relative to the --source directory)
+  // defaults to "."
+  "path": ".",
+  // specify path to required files (relative to the package path)
+  // where "npm-shrinkwrap.json" or "package-lock.json" is present.
+  // When both are present, npm-shrinkwrap.json takes precedence.
+}
+```
+
+## npm dependencies
+
+The "npm package" formats that Cachi2 can process are the following:
+```
+1. A folder containing a program described by a package.json file.
+2. A gzipped tarball containing (1.).
+3. A URL that resolves to (2.).
+4. A <name>@<version> that is published on the registry with (3.).
+5. A <name>@<tag> that points to (4.).
+6. A <name> that has a latest tag satisfying (5.).
+7. A git url that, when cloned, results in (1.).
+```
+
+Examples of (package.json) dependency formats:
+(For the full list of dependency formats with explanation, See [npm documentation](https://docs.npmjs.com/cli/v9/configuring-npm/package-json#dependencies)).
+
+<details>
+<summary>npm registries</summary>
+
+```jsonc
+{
+  "dependencies": {
+    "foo": "1.0.0 - 2.9999.9999",
+    "bar": ">=1.0.2 <2.1.2",
+    "baz": ">1.0.2 <=2.3.4",
+    "boo": "2.0.1",
+    ...
+  }
+}
+```
+</details>
+
+<details>
+<summary>URLs as Dependencies</summary>
+
+```jsonc
+{
+  "dependencies": {
+    "cli_bar": git+ssh://git@github.com:npm/cli.git#v1.0.27,
+    "cli_foo": git://github.com/npm/cli.git#v1.0.1
+  }
+}
+```
+</details>
+
+<details>
+<summary>GitHub URLs</summary>
+
+```jsonc
+{
+  "dependencies": {
+    "express": "expressjs/express",
+    "mocha": "mochajs/mocha#4727d357ea",
+    "module": "user/repo#feature\/branch"
+  }
+}
+```
+</details>
+
+
+<details>
+<summary>Local paths</summary>
+
+```jsonc
+{
+  "name": "baz",
+  "dependencies": {
+    "bar": "file:../foo/bar"
+  }
+}
+```
+</details>
+
+## Required files
+
+Cachi2 downloads dependencies explicitly declared in required files - package.json and package-lock.json.
+
+npm manages the package-lock.json file automatically. To make sure the file is up to date, you can use [npm install](https://docs.npmjs.com/cli/v9/commands/npm-install?v=true).
+
+#### Project example:
+<details>
+<summary>package.json</summary>
+
+```jsonc
+cat package.json
```

Comment:
Why the `cat` commands in the JSON examples?

Reply:
Removed